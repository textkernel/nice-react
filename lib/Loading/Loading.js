import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import props from '../props';

const Loading = ({ size, context, hidden }) => (
    <div className={ classNames('nice-loading', size, context, {
        hidden
    }) } />
);

Loading.propTypes = {
    size: PropTypes.oneOf(props.sizes),
    context: PropTypes.oneOf(props.contexts),
    hidden: PropTypes.bool
};

Loading.defaultProps = {
    size: 'small',
    context: 'brand',
    hidden: false
};

export default Loading;
