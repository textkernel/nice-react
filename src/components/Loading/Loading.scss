@import '../../styles/nice';

.nice-loading {
    border: .25rem solid rgba($color-black, .15);
    border-top-color: rgba($color-accent, 1);
    height: 24px;
    width: 24px;
    @include nice-animation(spin 1s linear 0s infinite);
    @include nice-border-radius(50%);
    @include nice-transition(border-top-color .5s);
    @extend %display-inline-block;

    @each $type in (brand, primary, info, good, warning, bad) {
        &.#{$type} {
            border-top-color: rgba(map-get($color-scheme, $type), 1);
        }
    }
    &.medium {
        border-width: .35rem;
        height: 48px;
        width: 48px;
        @include nice-animation($duration: 1.25s);
    }
    &.large {
        border-width: .45rem;
        height: 128px;
        width: 128px;
        @include nice-animation($duration: 1.5s);
    }
}
