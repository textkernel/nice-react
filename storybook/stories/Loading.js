import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, select } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';
import props from '../../src/props';

import { Loading } from 'nice-react';

const stories = storiesOf('Loading', module);

stories.addDecorator((story, context) => withInfo('')(story)(context));
stories.addDecorator(withKnobs);

stories.add('with default loader', () => (
    <Loading />
));

stories.add('with big size', () => (
    <Loading size="large" />
));

stories.add('with medium size and context "good"', () => (
    <Loading size="medium" context="good" />
));

stories.add('with dynamic properties', () => {
    const context = select('Context', props.contexts, props.contexts[0]);
    const size = select('Size', props.sizes, props.sizes[0]);
    const hidden = boolean('Hidden', false);

    return (<Loading
        size={ size }
        context={ context }
        hidden={ hidden }
    />);
});