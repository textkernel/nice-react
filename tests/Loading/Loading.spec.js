import React from 'react';
import Loading from '../../lib/Loading';

describe('<Loading />', () => {
  it('should render the Loading', () => {
    const wrapper = shallow(<Loading />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Loading with custom classes / styles', () => {
    const wrapper = shallow(<Loading className="test" style={{ display: 'none' }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the Loading with context', () => {
    const wrapper = shallow(<Loading context='good' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the Loading with different size', () => {
    const wrapper = shallow(<Loading size='large' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the Loading with size and context properties', () => {
    const wrapper = shallow(<Loading size='medium' context='bad' />);
    expect(wrapper).toMatchSnapshot();
  });
});
