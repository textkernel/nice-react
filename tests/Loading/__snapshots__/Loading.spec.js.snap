// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render Loading with custom classes / styles 1`] = `
<div
  className="nice-loading small brand test"
  style={
    Object {
      "display": "none",
    }
  }
/>
`;

exports[`<Loading /> should render the Loading 1`] = `
<div
  className="nice-loading small brand"
  style={null}
/>
`;

exports[`<Loading /> should render the Loading with context 1`] = `
<div
  className="nice-loading small good"
  style={null}
/>
`;

exports[`<Loading /> should render the Loading with different size 1`] = `
<div
  className="nice-loading large brand"
  style={null}
/>
`;

exports[`<Loading /> should render the Loading with size and context properties 1`] = `
<div
  className="nice-loading medium bad"
  style={null}
/>
`;
